### ISSUE TITLE
Angular animations spent too much time on startup (flushAnimations)

### ISSUE THREAD

**Issue by artaommahe** [angular/angular#33361]
# 🐞 bug report

### Affected Package

The issue is caused by package @angular/animations

### Description

On animation start there is a `TransitionAnimationEngine._flushAnimations` call that spent a lot of time on `writeStyleAttribute` and `camelCaseToDashCase` calls.

We have a custom select component with a complex expand animation (10 angular animation triggers) and its animation start is painfull.

From the real app
![image](https://user-images.githubusercontent.com/1944061/67419736-ef1cec00-f5d5-11e9-9abd-aa631d89bf0a.png)
![image](https://user-images.githubusercontent.com/1944061/67419847-1ecbf400-f5d6-11e9-86cd-f8d8b3b801ee.png)

From stackblitz example
![image](https://user-images.githubusercontent.com/1944061/67419973-618dcc00-f5d6-11e9-8625-6952878cd7f1.png)
![image](https://user-images.githubusercontent.com/1944061/67420012-7a967d00-f5d6-11e9-9350-9c3a163195cd.png)

## 🔬 Minimal Reproduction

simplified case
https://stackblitz.com/edit/angular-animations-slow-startup?file=src/app/app.component.ts

## 🌍  Your Environment

**Angular Version:**
8.x.x

**Comment by blidblid** [comment]
This caused _massive_ performance issues in our team. When loading our UI, 40% of our was spent on writeStyleAttribute.

If you're having issues with this, look to see if you have a `process` on `window`. Our webpack was configured to add a `process` on `window`, meaning that this evaluated to true in `@angular/animations`:

```
isNode() {
  return process !== 'undefined';
}
```

It made Angular treat our animations as if they were on a node server and the regexp `camelCaseToDashCase` delayed our first meaningful by a few seconds. 

This also means that a third party could assign a process and tax Angular performance. If I assign `process` in [mat-expansion](https://material.angular.io/components/expansion/examples), performance plummets:


### With assigned process
`process === 'bye bye angular performance';`

![image](https://user-images.githubusercontent.com/30099577/70803317-11e8a700-1db4-11ea-9f50-3ea2cec4f372.png)

### Without assigned process
`process === undefined;`

![image](https://user-images.githubusercontent.com/30099577/70803362-2d53b200-1db4-11ea-9ae8-87f8c1d89bd8.png)

**Comment by artaommahe** [comment]
In our case there is no `window.process` variable

**Comment by blidblid** [comment]
In our case, we actually couldn't find `process` on `window` in dev-tools. But we saw it when we logged in `isNode()` in `node_modules\@angular\animations\*your-target*\browser.js`. I'd go there and have a look.

**Comment by artaommahe** [comment]
yep, added debug inside `isNode` shows `true` value, at least in dev mode

looks like there is a pr already https://github.com/angular/angular/pull/32903

**Comment by artaommahe** [comment]
Checked same animation from real app in dev mode with current isNode check and with manually changed to `return false` inside `animations/browser.js`

`isNode === true`
![image](https://user-images.githubusercontent.com/1944061/70807446-9b0cd780-1dce-11ea-89dd-911975666320.png)

`isNode === false`
![image](https://user-images.githubusercontent.com/1944061/70807425-9516f680-1dce-11ea-99ae-632667d5c44e.png)

So it's 40ms vs 5ms difference, 8 times faster.

@gkalpak @matsko can we have it fixed? there is a huge performance drop now

**Comment by blidblid** [comment]
Before a change to `isNode`, you could modify your webpack config:

```
const config: webpack.Configuration = {
  ...,
  node: {
    process: false
  }
};
```

**Comment by artaommahe** [comment]
Used this hack now https://github.com/angular/angular/issues/24769#issuecomment-527778131 cause process.env is used by some other libs.

**Comment by swseverance** [comment]
This appears to have been fixed by [this PR](https://github.com/angular/angular/pull/35134)

**Comment by JoostK** [comment]
Indeed, this code is specific to NodeJS environments which was incorrectly executing in the browser. Resolved in #35134.
